### ISSUE TITLE
2019-8-9

### ISSUE THREAD

**Issue by ssfc** [ssfc/wangdao#77]
281.Look for keras; replace importlibrary with import ID; syntax first generate symbols, then use physical address to replace the symbol; The name compiler is primarily used for programs that translate source code from a high-level programming language to a lower level language; do preprocessor has name of input and output? No; (2019-7-4)
282.What is &#38; translate source is used in btn_translate; change the button in analyzer.html, done; look for keras first appeared in Andrew Ng’s course; find it; Convolutional Neural Networks/Week 2/Keras Tutorial - The Happy House (not graded); check keras on zhihu; continue Andrew Ng sequence model course week 2 and week 3; make sure the similarity is between 0 and 1; change SelectEntry to SelectSentence; done; rename select_compound as select_sentence; done; change compounds.php to show_all_sentences.php, done; (2019-7-4)
283.modify sort_similarity.php so that similarity value is between 0 and 1; move out sql; done; delete group; we can use word-embedding to implement group; take element_group out of the inner for loop; done; (2019-7-4)
284.Read our translate_source file and understand it; change ElementID to WordID; done; word also fit POS well; delete compiler.php; 
285.The similarity between AB and BA is different, should we change the situation? 
286.Try to run LSTM to make it work; done; upload them; try to modify dinosaur file; what is unique characters; it is about text generation; what is sample; figure out the usage of each function; python set function; example for 26 letters, there is \n; Now assume that your model is trained. You would like to generate new text (characters); y becomes input x through sampling; clip, sample, optimize; stochastic gradient descent; strange, how they get loss from run forward; we must know the function in it; (2019-7-5)
287.What is cross-entropy; 交叉熵损失函数经常用于分类问题中，特别是在神经网络做分类问题时，也经常使用交叉熵作为损失函数，此外，由于交叉熵涉及到计算每个类别的概率，所以交叉熵几乎每次都和softmax函数一起出现; what is convex function; 为什么数学概念中，将凸起的函数称为凹函数; log function is cross-entropy function; check softmax on Zhihu; softmax用于多分类过程中，它将多个神经元的输出，映射到（0,1）区间内，可以看成概率来理解，从而来进行多分类; 用了softmax函数之后，梯度求导过程非常非常方便; we are glad to find util.py, use it to implement out function; success; (2019-7-5)
288.Let us try to change dino.txt to aaa, bbb, ccc and check the result; yes it works; (2019-7-5)
289.Is the format correct? Give up; (2019-7-6)
